OPHD/Technology/ResearchTracker.cpp

```diff
@@ -12,12 +12,7 @@ void ResearchTracker::startResearch(int techId, int progress, int assigned)
 		throw std::runtime_error("Can't update a technology that's already researched.");
 	}
 
-	auto emplacedItem = mTechnologiesBeingResearched.emplace(techId, ResearchProgress{progress, assigned});
-	if (!emplacedItem.second)
-	{
-		(*emplacedItem.first).second = {progress, assigned};
-	}
-
+	mTechnologiesBeingResearched.insert_or_assign(techId, ResearchProgress{progress, assigned});

```

Comment:
Didn't even know this was available... because it exactly solves the original code.

Doesn't help that cppreference is _really_ hard to read.

Reply:
Admittedly it does take a bit of effort to learn how to read CppReference. They can be rather pedantic, though very detailed and accurate. They do sometimes have a bit of a high level explanation of stuff, though mostly they're going into the minute details, which are not always easy to read or understand. It's not quite as bad as the legalese you'd find in the C++ Standard, but it's a good way there. I find it's a good intermediate source of documentation.